src/ipahealthcheck/ipa/certs.py

```diff
@@ -583,6 +584,125 @@ def check(self):
+            return
+        if cacert != entry['CACertificate'][0]:
+            yield Result(self, constants.ERROR,
+                         msg=('CA Certificate entry in LDAP does not '
+                              'match %s' % paths.IPA_CA_CRT))
+            return
+
+        # Ensure NSS DBs have matching CA certs for /etc/ipa/ca.crt
+        serverid = realm_to_serverid(api.env.realm)
+        dspath = paths.ETC_DIRSRV_SLAPD_INSTANCE_TEMPLATE % serverid
+        cacertds_ok = yield from match_cacert_and_db(self, cacert, dspath)
+        cacertnss_ok = yield from match_cacert_and_db(self, cacert,
+                                                      paths.IPA_NSSDB_DIR)
+
+        if cacertds_ok and cacertnss_ok:
+            yield Result(self, constants.SUCCESS)
+
+
+@registry
+class IPADogtagCertsMatchCheck(IPAPlugin):
+    """
+    Check if dogtag certs present in both NSS DB and LDAP match
+    """
+    requires = ('dirsrv',)
+
+    @duration
+    def check(self):
+        if not self.ca.is_configured():
+            logger.debug('CA is not configured, skipping connectivity check')
+            return
+
+        def match_ldap_nss_cert(plugin, ldap, db, cert_dn, attr, cert_nick):
+            try:
+                entry = ldap.get_entry(cert_dn)
+            except errors.NotFound:
+                yield Result(plugin, constants.ERROR,
+                             msg='%s entry not found in LDAP' % cert_dn)
+                return False
+            try:
+                nsscert = db.get_cert_from_db(cert_nick)
+            except Exception as e:
+                yield Result(plugin, constants.ERROR,
+                             error=str(e),
+                             msg=('Unable to load %s certificate:'
+                                  '{error}' % cert_nick))
+                return False
+            ldapcert = entry[attr][0]
+            if nsscert != ldapcert:
+                yield Result(plugin, constants.ERROR,
+                             msg=('\'%s\' certificate in NSS DB does not '
+                                  'does not match entry in LDAP' % cert_nick))
+                return False
+            return True
+
+        db = certs.CertDB(api.env.realm, paths.PKI_TOMCAT_ALIAS_DIR)
+        dn = DN('uid=pkidbuser,ou=people,o=ipaca')
+        nick = 'subsystemCert cert-pki-ca'
+        subsystem_ok = yield from match_ldap_nss_cert(self, self.conn,
+                                                      db, dn,
+                                                      'userCertificate', nick)
+        dn = DN('cn=%s IPA CA' % api.env.realm,
+                'cn=certificates,cn=ipa,cn=etc',
+                api.env.basedn)
+        nick = 'caSigningCert cert-pki-ca'
+        casigning_ok = yield from match_ldap_nss_cert(self, self.conn,
+                                                      db, dn, 'CACertificate',
+                                                      nick)
+        if subsystem_ok and casigning_ok:
```

Comment:
The Audit, OCSP and subystem certificates should also be checked. And if there is a KRA then those as well.